core/src/main/java/feign/ReflectiveFeign.java

```diff
@@ -160,21 +167,16 @@ public String toString() {
     protected final MethodMetadata metadata;
     private final Map<Integer, Expander> indexToExpander = new LinkedHashMap<Integer, Expander>();
 
-    private BuildTemplateByResolvingArgs(MethodMetadata metadata) {
+    private BuildTemplateByResolvingArgs(MethodMetadata metadata, Factory paramExpanderFactory) {
       this.metadata = metadata;
+      checkNotNull(paramExpanderFactory, "paramExpanderFactory");
       if (metadata.indexToExpanderClass().isEmpty()) {
         return;
       }
       for (Entry<Integer, Class<? extends Expander>> indexToExpanderClass : metadata
```

Comment:
ex move the instantiation logic to DefaultContract, and just assign `this.indexToExpander = md.indexToExpander`
A spring contract wouldn't use newInstance, but we don't need a factory type either.